kedro/io/data_catalog.py

```diff
@@ -1040,7 +1048,10 @@ def load(self, ds_name: str, version: str | None = None) -> Any:
         dataset = self.get(ds_name, version=load_version)
 
         if dataset is None:
-            error_msg = f"Dataset '{ds_name}' not found in the catalog"
+            error_msg = f"Dataset '{ds_name}' not found in the catalog . Did you mean 'suggestions'?"
```

Comment:
I'm not sure this logic is entirely correct. If there are no suggestions you see "Did you mean 'suggestions'" in the error message, which doesn't make sense. I'd expect the "Did you mean ..." part to only be in the message when there are actual suggestions to give.